lib/ansible/module_utils/ec2.py

```diff
@@ -104,9 +104,13 @@ def found(response_code, catch_extra_error_codes=None):
 def boto3_conn(module, conn_type=None, resource=None, region=None, endpoint=None, **params):
     try:
         return _boto3_conn(conn_type=conn_type, resource=resource, region=region, endpoint=endpoint, **params)
-    except ValueError:
-        module.fail_json(msg='There is an issue in the code of the module. You must specify either both, resource or client to the conn_type '
-                             'parameter in the boto3_conn function call')
+    except ValueError as e:
```

Comment:
I'm very much in favor of fixing this handling here, so when we discover it's lacking it can be fixed in just one place. I think this needs to also catch botocore.exceptions.PartialCredentialsError.

Reply:
You might be right - how do I generate such an error?

Reply:
If you pass an access key or secret key (but not both of them) to the module directly without adequate fall back environment variables that happens. I assume this happens with inadequate environment variables too probably.